### PR TITLE
Set up API

### PR DESCRIPTION
**Pull request by karelvanoordt**
- Made JavaScript asynchronous 
- Used API to get and post data into the app.
- Categorized functions in different files.

### CODE REVIEW

File: src/leaderboard.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import './style.css';
+
+export default function currentLeaderboard(result) {
```
Comment (helman101) [inline src/leaderboard.js]: - [x] Please avoid using the `function` keyword, instead use `arrow functions`, this is part of the module to make you a master in ES6 syntax 😄

File: src/index.js [hunk 2]
```diff
@@ -1,18 +1,24 @@
...
-  item.textContent = `${element.name}:  ${element.score}`;
-  container.appendChild(item);
+submitBtn.addEventListener('click', async (e) => {
+  e.preventDefault();
+  const userInputF = userInput.value;
+  const scoreInputF = scoreInput.value;
+  await postScore(userInputF, scoreInputF);
+});
+
+function updateLeaderboard() {
```
Comment (helman101) [inline src/index.js]: - [x] Please avoid using the `function` keyword, instead use `arrow functions`, this is part of the module to make you a master in ES6 syntax 😄

File: src/index.js [hunk 3]
```diff
@@ -1,18 +1,24 @@
...
+  getScore().then((scores) => {
```
Comment (helman101) [inline src/index.js]: - [x] Please use `async` & `await` here, you can call this function like this:
    ```javascript
  const updateLeaderboard = () => {
      const scores = await getScore()
      currentLeaderboard(scores);
  }
    ```

File: src/gameapi.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+const gameScoreUrl = 'https://us-central1-js-capstone-backend.cloudfunctions.net/api/games/0VAg54chbZRX31kPbycC/scores';
```
Comment (helman101) [inline src/gameapi.js:1]: - [x] Currently, your page is not working, I see that you are calling the API but the page do not display any information that I submit as you can see in the gif below, please fix this issue 🤔

  leaderboard

**Review by helman101 — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 🔴 

Hi @karelvanoordt  👋, 

Nice job so far with this project 😄, but there are still some changes to do.

## Required Changes ♻️

- [ ] _Check the comments under the review._

Cheers and Happy coding!👏👏👏

gif

If anything is not 100% clear please leave your questions or comments in the PR thread, we will be glad to help you 😄.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/gameapi.js (hunk 4 above)
Comment (karelvanoordt) [reply src/gameapi.js:1]: Thanks, Andres for the comments. I actually changed a few things before submitting so you made me notice the issues.

Comment (helman101) [reply src/gameapi.js:1]: - [ ] Nice job so far, but there is still a small issue, when i start the app for the first time I'm not able to see any o the scores, only when i hit the refresh button the records are displayed, please fix this 😄

**Review by helman101 — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 🔴 

Hi @karelvanoordt  👋, 

Great work on making the changes requested by a previous reviewer 👏 👏👏
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## To Highlight 🌟 

- Nice job replacing the functions for arrow functions 😄 
- Nice job adding using `async` & `await` 🚀 

## Required Changes ♻️

- [ ] _Check the comments under the review._

Cheers and Happy coding!👏👏👏

gif

If anything is not 100% clear please leave your questions or comments in the PR thread, we will be glad to help you 😄.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by BrianSammit — APPROVED** [review]
# STATUS: APROVED :green_circle: :clap: 

Hello @karelvanoordt, 
Your project is complete!  There is nothing else to say other than... Great job 👏 
Congratulations! 🎉 

Now you can merge this branch and move on :1st_place_medal: 
HAPPY CODING AND GOOD LUCK WITH THE COMMING ONES :crossed_fingers: :four_leaf_clover: 

well-1

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**